k2/csrc/array_ops.cu

```diff
@@ -304,10 +304,39 @@ Array1<int32_t> GetCounts(ContextPtr c, const int32_t *src_data,
         nullptr, temp_storage_bytes, src_data, ans_data, n + 1, 0, n, src_dim,
         c->GetCudaStream()));  // The first time is to determine temporary
                                // device storage requirements.
-    Array1<int8_t> d_temp_storage(c, temp_storage_bytes);
-    K2_CHECK_CUDA_ERROR(cub::DeviceHistogram::HistogramEven(
-        d_temp_storage.Data(), temp_storage_bytes, src_data, ans_data, n + 1, 0,
-        n, src_dim, c->GetCudaStream()));
+
+    constexpr std::size_t kTreshold = (static_cast<std::size_t>(1) << 33);
+
+    if (temp_storage_bytes < kTreshold) {
+      RegionPtr temp_storage = NewRegion(c, temp_storage_bytes);
+      K2_CHECK_CUDA_ERROR(cub::DeviceHistogram::HistogramEven(
+          temp_storage->data, temp_storage_bytes, src_data, ans_data, n + 1, 0,
+          n, src_dim, c->GetCudaStream()));
+    } else {
+      // split the array and do a recursive call
```

Comment:
Relates to https://github.com/NVIDIA/cub/issues/288

Reply:
kThreshold.

Reply:
Fixed. Thanks